__tests__/src/index.spec.ts

```diff
@@ -751,6 +751,41 @@ describe("flatpickr", () => {
         expect(fp.selectedDates[0]).toEqual(new Date("2020-01-22T00:00:00"));
         expect(fp._input.value).toBe("22 January 2020");
       });
+
+      it("should be updated correctly via mouse events when altInput is enabled #2653", () => {
+        createInstance({
+          allowInput: true,
+          altInput: true,
+          altFormat: "j F Y H:i",
+          defaultDate: "2020-01-22 03:04",
+          enableTime: true,
+        });
+
+        expect(fp.selectedDates[0]).toEqual(new Date("2020-01-22T03:04:00"));
+        simulate("focus", fp._input);
+        simulate("mousedown", window.document.body, { which: 1 }, CustomEvent);
+        expect(fp.isOpen).toBe(false);
+        expect(fp._input.value).toBe("22 January 2020 03:04");
+      });
+
+      it("should trigger onChange only once", () => {
```

Comment:
This test is for the issue mentioned here:
https://github.com/flatpickr/flatpickr/pull/2601#issuecomment-1063960360